dev/src/lobster/flat_gltf.h

```diff
@@ -0,0 +1,789 @@
+
+namespace fgl {
+
+bool verbose = true;
+
+#ifndef _VERBOUT_
+#define _VERBOUT_                                                              \
+  if (verbose)                                                                 \
+  std::cout
+#endif
+
+#ifndef _ERROUT_
+#define _ERROUT_ std::cerr
+#endif
+
+typedef struct {
+  bool success;
+  GLuint texture;
+  GLsizei width;
+  GLsizei height;
+  unsigned int channels;
+  bool mipmap;
+} Texture;
+
+Texture newTexture = {false, 0, 0, 0, 0, false};
+
+Texture drawTexture0 = {false, 0, 0, 0, 0, false};
+
+float4x4 texMatrix = float4x4_1;
+
+// TODO: pbr_anim shader
+typedef struct {
+  GLfloat Position[4]; // vec4 position is not necessary; GLSL auto-converts vec3 to vec4
+  GLfloat Normal[3];
+  GLfloat UV[2];
+  GLfloat Color[4]; // not used in glTF but whatever, it's in the phong_anim shader
+  GLfloat Weights[4];
+  GLfloat Indices[4]; // floating-point integers. thanks GL 2.0, aren't we past that?
+} Vertex;
+
+Vertex newVertex = {
+  {0.0f, 0.0f, 0.0f, 0.0f},
+  {0.0f, 0.0f, 0.0f},
+  {0.0f, 0.0f},
+  {1.0f, 1.0f, 1.0f, 1.0f},
+  {0.0f, 0.0f, 0.0f, 0.0f},
+  {0.0f, 0.0f, 0.0f, 0.0f}
+};
+
+typedef GLuint Index;
+
+typedef struct {
+  bool success;
+  GLuint vao[3]; // VAO state, vertex buffer, index buffer
+  GLsizei indexCount;
+} Mesh;
+
+Mesh newMesh = {false, {0, 0, 0}, 0};
+
+Mesh LoadMesh(std::vector<Vertex> &vertices, std::vector<Index> &indices,
+              bool streaming = false) {
+  Mesh mesh = newMesh;
+
+  _VERBOUT_ << (streaming ? "STREAMING " : "") << "VERTEX ARRAY OBJECT"
+            << std::endl;
+
+  // generate a VAO (1 per mesh)
+  glGenVertexArrays(1, &mesh.vao[0]);
+  // switch graphics memory to the VAO
+  glBindVertexArray(mesh.vao[0]);
+
+  _VERBOUT_ << "VAO address:           " << mesh.vao[0] << std::endl;
+
+  // turn on 6 attribute arrays
+  glEnableVertexAttribArray(0); // Position
+  glEnableVertexAttribArray(1); // Normal
+  glEnableVertexAttribArray(2); // UV
+  glEnableVertexAttribArray(3); // Color (not used)
+  glEnableVertexAttribArray(4); // Weights
+  glEnableVertexAttribArray(5); // Indices
+
+  // generate a vertex buffer and an index buffer
+  glGenBuffers(2, &mesh.vao[1]);
+
+  _VERBOUT_ << "Vertex buffer address: " << mesh.vao[1] << std::endl;
+  _VERBOUT_ << "Index buffer address:  " << mesh.vao[2] << std::endl;
+
+  // switch to the vertex buffer
+  glBindBuffer(GL_ARRAY_BUFFER, mesh.vao[1]);
+  // upload the vertices, passing the size in bytes
+  glBufferData(GL_ARRAY_BUFFER, vertices.size() * sizeof(Vertex), &vertices[0],
+               streaming ? GL_STREAM_DRAW : GL_STATIC_DRAW);
+  _VERBOUT_ << "Number of vertices: " << vertices.size() << std::endl;
+  _VERBOUT_ << "Size of Vertex:     " << sizeof(Vertex) << std::endl;
+
+  // perhaps an inconsistent use of data here but I'm not sure what to do
+  // sizeof returns a size_t which can be cast to anything since it's just a
+  // number
+  std::size_t pointer = 0;
```

Comment:
Ah apologies, I never added a clang format file to this repo, and you formatted with the default style. I just added such a file. Can you run it again after you rebased against master?

Reply:
I have no idea how to do that other than overwriting files and opening a new PR. I've had about enough of Git and GitHub.

Reply:
Sorry, I can't fix that for you.

And try `git clang-format`, it only formats the code in your PR.

Reply:
1git: 'clang-format' is not a git command. See 'git --help'.`

git 2.17.1

Reply:
`git: 'clang-format' is not a git command. See 'git --help'.`

git 2.17.1

Rebased and all. Oh man, this site is a mess.

Reply:
Again, just because you get an error doesn't mean the issue is resolved. I am sure you can find out how to either install clang-format for git, or just run clang-format manually on your own files.

Reply:
`git clang-format-7` was needed on my machine. Resolved now.